gateway/gateway.go

```diff
@@ -2,36 +2,48 @@ package gateway
 
 import (
 	"context"
+	"io"
 	"net/http"
 	"sort"
 
-	coreiface "github.com/ipfs/interface-go-ipfs-core"
-	path "github.com/ipfs/interface-go-ipfs-core/path"
+	cid "github.com/ipfs/go-cid"
+	ipld "github.com/ipfs/go-ipld-format"
+	"github.com/ipfs/go-libipfs/files"
+	iface "github.com/ipfs/interface-go-ipfs-core"
+	options "github.com/ipfs/interface-go-ipfs-core/options"
+	"github.com/ipfs/interface-go-ipfs-core/path"
 )
 
 // Config is the configuration that will be applied when creating a new gateway
 // handler.
 type Config struct {
-	Headers  map[string][]string
-	Writable bool
+	Headers map[string][]string
 }
 
-// NodeAPI defines the minimal set of API services required by a gateway handler
+// NodeAPI defines the minimal set of API services required for a read-only
+// gateway handler. These are, for the most part, subsets of the core interface.
+
 type NodeAPI interface {
-	// Unixfs returns an implementation of Unixfs API
-	Unixfs() coreiface.UnixfsAPI
+	// GetUnixFs returns a read-only handle to a file tree referenced by a path.
+	GetUnixFs(context.Context, path.Path) (files.Node, error)
+
+	// LsUnixFs returns the list of links in a directory.
+	LsUnixFs(context.Context, path.Path, ...options.UnixfsLsOption) (<-chan iface.DirEntry, error)
+
+	// GetBlock attempts to resolve the path and return a reader for data in the block.
+	GetBlock(context.Context, path.Path) (io.Reader, error)
```

Comment:
nits:
- Since we have a separate `ResolvePath`, does it make sense to change this to require  [`Resolved`](https://github.com/ipfs/interface-go-ipfs-core/blob/master/path/path.go#L44) path instead?
- It returns a `BlockReader`, not a block bytes

```suggestion
	BlockReader(context.Context, path.Resolved) (io.Reader, error)
```